tests/Unit/ProtoTests/BasicOneof/out/src/Request/SupplementaryDataOneof.php

```diff
@@ -67,4 +67,207 @@ class SupplementaryDataOneof
+     *
+     * @param mixed $extraIndex The new value of this oneof.
+     */
+    public function setExtraIndex($extraIndex)
+    {
+        $this->extraIndex = $extraIndex;
+        $this->selectedOneofFieldName = 'extra_index';
+        return $this;
+    }
+
+    /**
+     * Sets this oneof to extra_double and updates its value.
+     *
+     * @param mixed $extraDouble The new value of this oneof.
+     */
+    public function setExtraDouble($extraDouble)
+    {
+        $this->extraDouble = $extraDouble;
+        $this->selectedOneofFieldName = 'extra_double';
+        return $this;
+    }
+
+    /**
+     * Sets this oneof to extra_float and updates its value.
+     *
+     * @param mixed $extraFloat The new value of this oneof.
+     */
+    public function setExtraFloat($extraFloat)
+    {
+        $this->extraFloat = $extraFloat;
+        $this->selectedOneofFieldName = 'extra_float';
+        return $this;
+    }
+
+    /**
+     * Sets this oneof to extra_bool and updates its value.
+     *
+     * @param mixed $extraBool The new value of this oneof.
+     */
+    public function setExtraBool($extraBool)
+    {
+        $this->extraBool = $extraBool;
+        $this->selectedOneofFieldName = 'extra_bool';
+        return $this;
+    }
+
+    /**
+     * Returns true if this oneof is set to the field extra_description.
+     */
+    public function isExtraDescription()
+    {
+        return $this->selectedOneofFieldName === 'extra_description';
+    }
+
+    /**
+     * Returns true if this oneof is set to the field extra_summary.
+     */
+    public function isExtraSummary()
+    {
+        return $this->selectedOneofFieldName === 'extra_summary';
+    }
+
+    /**
+     * Returns true if this oneof is set to the field extra_request.
+     */
+    public function isExtraRequest()
+    {
+        return $this->selectedOneofFieldName === 'extra_request';
+    }
+
+    /**
+     * Returns true if this oneof is set to the field extra_index.
+     */
+    public function isExtraIndex()
+    {
+        return $this->selectedOneofFieldName === 'extra_index';
+    }
+
+    /**
+     * Returns true if this oneof is set to the field extra_double.
+     */
+    public function isExtraDouble()
+    {
+        return $this->selectedOneofFieldName === 'extra_double';
+    }
+
+    /**
+     * Returns true if this oneof is set to the field extra_float.
+     */
+    public function isExtraFloat()
+    {
+        return $this->selectedOneofFieldName === 'extra_float';
+    }
+
+    /**
+     * Returns true if this oneof is set to the field extra_bool.
+     */
+    public function isExtraBool()
+    {
+        return $this->selectedOneofFieldName === 'extra_bool';
+    }
+
+    /**
+     * Returns $this->extraDescription if this oneof is set to the field
+     * extra_description, null otherwise.
+     */
+    public function getExtraDescription()
+    {
+        return $this->isExtraDescription() ? $this->extraDescription : null;
```

Comment:
If you prefer, there is a shorthand operator that can be used:

```suggestion
        return $this->isExtraDescription() ?: $this->extraDescription;
```